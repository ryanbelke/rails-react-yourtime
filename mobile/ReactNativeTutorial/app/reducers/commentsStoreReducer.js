import { fromJS } from 'immutable';

const CREATE = 'COMMENTS_STORE:CREATE';
const REMOVE = 'COMMENTS_STORE:REMOVE';
const SET_LOADING = 'COMMENTS_STORE:SET_LOADING';

export const initialState = fromJS({
  meta: {
    loading: false,
  },
});

const create = (state, action) => state.merge(action.entities);
const remove = (state, action) => state.delete(action.id);
const setLoading = (state, action) => state.setIn(['meta', 'loading'], action.loading);

export default (state = initialState, action) => {
  switch (action.type) {
    case CREATE:
      return create(state, action);
    case REMOVE:
      return remove(state, action);
    case SET_LOADING:
      return setLoading(state, action);
    default:
      return state;
  }
};

const createComments = (entities) => ({ type: CREATE, entities });
const removeComment = (id) => ({ type: REMOVE, id });
const setLoadingComments = (loading) => ({ type: SET_LOADING, loading });

export const actions = {
  createComments,
  removeComment,
  setLoadingComments,
};
