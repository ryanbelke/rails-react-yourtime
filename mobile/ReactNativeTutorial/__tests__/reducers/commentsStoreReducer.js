import { fromJS } from 'immutable';
import commentsStoreReducer, { actions, initialState } from 'ReactNativeTutorial/app/reducers/commentsStoreReducer';

const sampleData = {
  1: { id: 1, author: 'Alexey', text: 'Random comment' },
  2: { id: 2, author: 'Justin', text: 'Another random comment' },
};

describe('commentsStoreReducer', () => {
  describe('createComments', () => {
    it('adds comments to the store', () => {
      expect(
        commentsStoreReducer(
          initialState,
          actions.createComments(sampleData))
      ).toMatchSnapshot();
    });
  });

  describe('removeComment', () => {
    it('removes comment by id', () => {
      expect(
        commentsStoreReducer(initialState.merge(sampleData), actions.removeComment('1'))
      ).toMatchSnapshot();
    });
  });

  describe('setLoadingComments', () => {
    it('sets the loading meta information', () => {
      const trueState = commentsStoreReducer(initialState, actions.setLoadingComments(true));
      const falseState = commentsStoreReducer(trueState, actions.setLoadingComments(false));
      expect(trueState).toMatchSnapshot();
      expect(falseState).toMatchSnapshot();
    });
  });
});

