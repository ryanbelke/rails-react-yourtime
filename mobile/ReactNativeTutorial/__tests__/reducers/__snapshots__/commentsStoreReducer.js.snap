exports[`commentsStoreReducer createComments adds comments to the store 1`] = `
Object {
  "1": Object {
    "author": "Alexey",
    "id": 1,
    "text": "Random comment",
  },
  "2": Object {
    "author": "Justin",
    "id": 2,
    "text": "Another random comment",
  },
  "meta": Object {
    "loading": false,
  },
}
`;

exports[`commentsStoreReducer removeComment removes comment by id 1`] = `
Object {
  "2": Object {
    "author": "Justin",
    "id": 2,
    "text": "Another random comment",
  },
  "meta": Object {
    "loading": false,
  },
}
`;

exports[`commentsStoreReducer setLoadingComments sets the loading meta information 1`] = `
Object {
  "meta": Object {
    "loading": true,
  },
}
`;

exports[`commentsStoreReducer setLoadingComments sets the loading meta information 2`] = `
Object {
  "meta": Object {
    "loading": false,
  },
}
`;
